trulens_eval/examples/snowflake_demo/retrieve.py

```diff
@@ -36,3 +36,38 @@ def retrieve(self, query: str):
         contents = [json.loads(t.page_content) for t in nodes]
         texts = [tc.get("text") for tc in contents]
         return texts
+
+
+class CortexSearchRetriever:
+
+    def __init__(self, limit_to_retrieve: int = 4):
+        self._limit_to_retrieve = limit_to_retrieve
+        connection_parameters = {
+            "account": os.environ["SF_ACCOUNT"],
+            "user": os.environ["SF_USER"],
+            "password": os.environ["SF_PASSWORD"],
+            "role": os.environ["SF_ROLE"],
+            "warehouse": os.environ["SF_WAREHOUSE"],
+        }
+        session = Session.builder.configs(connection_parameters).create()
+        root = Root(session)
+        self._cortex_search_service = root.databases[
```

Comment:
what's the reason for using `Root` here to traverse the SF databases if we are only using the single specified DB and SCHEMA?

couldn't we just pass `SF_DB_NAME` and `SF_SCHEMA` into the connection_parameters?

Reply:
oh I was just following what the cortex search doc said to do:
https://docs.snowflake.com/LIMITEDACCESS/cortex-search/query-cortex-search-service#label-cortex-search-query-syntax-python
TBH, the service is kinda brittle so I figured I should just follow what they said exactly haha.